### PR TITLE
PLANNER-565 - Best solution test coverage extension

### PR DESCRIPTION
**Pull request by sutaakar**
@ge0ffrey FYI, I put here that change I sent you in PR.

### CODE REVIEW

**Comment by sutaakar** [comment]
@psiroky I have noticed that Kie server integration tests weren't invoked here. Is it some configuration issue?

**Comment by psiroky** [comment]
jenkins retest this (@sutaakar yes, it was a configuration issue, fixed by merging #423)

**Comment by mswiderski** [comment]
do we still need this/plan to merge?

**Comment by sutaakar** [comment]
@ge0ffrey what do you think about this test update?

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-optaplanner/src/test/java/org/kie/server/integrationtests/optaplanner/OptaplannerIntegrationTest.java [hunk 1]
```diff
@@ -265,8 +265,10 @@ public void testGetBestSolution() throws Exception {
         assertNotNull(score);
         assertTrue(solution.getScore().isSolutionInitialized());
         assertTrue(score.getHardScore() <= 0);
+        assertTrue(score.getHardScore() > -1000);
         // A soft score of 0 is impossible because we'll always need at least 1 computer
         assertTrue(score.getSoftScore() < 0);
+        assertTrue(score.getSoftScore() > -1000000);
```
Comment (ge0ffrey) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-optaplanner/src/test/java/org/kie/server/integrationtests/optaplanner/OptaplannerIntegrationTest.java]: I wouldn't do these 2 lines: we can't really guarantee that the first initialized score will be above -1k hard and -1m soft. That's why I didn't retain them from the original test.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-optaplanner/src/test/java/org/kie/server/integrationtests/optaplanner/OptaplannerIntegrationTest.java [hunk 2]
```diff
@@ -275,6 +277,7 @@ public void testGetBestSolution() throws Exception {
         for(Object process : processList) {
             Object computer = valueOf(process, "computer");
             assertNotNull(computer);
+            assertTrue(computerList.contains(computer));
```
Comment (ge0ffrey) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-optaplanner/src/test/java/org/kie/server/integrationtests/optaplanner/OptaplannerIntegrationTest.java]: This should actually be stronger than contains (which allows equals but not same objects too). The same instance should be in the computerList. But there's an open bug for JAXB (not for XStream) that that's not the case (see todo about @XmlID and @XmlIDREF).

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-optaplanner/src/test/filtered-resources/kjars-sources/cloudbalance/src/main/java/org/kie/server/testing/CloudComputer.java [hunk 3]
```diff
@@ -74,4 +76,30 @@ public int getMultiplicand() {
         return cpuPower * memory * networkBandwidth;
     }
 
+    public int hashCode() {
+        return new HashCodeBuilder()
+                .append(id)
```
Comment (ge0ffrey) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-optaplanner/src/test/filtered-resources/kjars-sources/cloudbalance/src/main/java/org/kie/server/testing/CloudComputer.java]: I'd argue that if the id is equals, then the computers are equal... (presuming the id isn't null of course).

Comment (sutaakar) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-optaplanner/src/test/filtered-resources/kjars-sources/cloudbalance/src/main/java/org/kie/server/testing/CloudComputer.java]: True, if we can presume that.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-optaplanner/src/test/java/org/kie/server/integrationtests/optaplanner/OptaplannerIntegrationTest.java (hunk 1 above)
Comment (sutaakar) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-optaplanner/src/test/java/org/kie/server/integrationtests/optaplanner/OptaplannerIntegrationTest.java]: Right, will remove them.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-optaplanner/src/test/java/org/kie/server/integrationtests/optaplanner/OptaplannerIntegrationTest.java (hunk 2 above)
Comment (sutaakar) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-optaplanner/src/test/java/org/kie/server/integrationtests/optaplanner/OptaplannerIntegrationTest.java]: I put a note there to update the test when @XmlID will be implemented.

**Comment by ge0ffrey** [comment]
+1 to merge.

**Pull request merged**